src/pages/instances/forms/CreateInstanceFromSnapshotForm.tsx

```diff
@@ -0,0 +1,308 @@
+import { fetchClusterMembers } from "api/cluster";
+import { Link } from "react-router-dom";
+import { instanceNameValidation, truncateInstanceName } from "util/instances";
+import { fetchProjects } from "api/projects";
+import { LxdDiskDevice } from "types/device";
+import InstanceLink from "pages/instances/InstanceLink";
+import { useEventQueue } from "context/eventQueue";
+
+interface Props {
+  instance: LxdInstance;
+  snapshot: LxdInstanceSnapshot;
+  close: () => void;
+}
+
+export interface CreateInstanceFromSnapshotValues {
+  instanceName: string;
+  targetProject: string;
+  targetClusterMember?: string;
+  targetStoragePool: string;
+  stateful: boolean;
+}
+
+export const instanceFromSnapshotPayload = (
+  values: CreateInstanceFromSnapshotValues,
+  instance: LxdInstance,
+  snapshot: LxdInstanceSnapshot,
+) => {
+  const basePayload: Record<string, unknown> = {
+    name: values.instanceName,
+    description: instance.description,
+    architecture: instance.architecture,
+    ephemeral: snapshot.ephemeral,
+    mode: "pull",
+    devices: {
+      ...instance.devices,
+      root: {
+        path: "/",
+        type: "disk",
+        pool: values.targetStoragePool,
+      },
+    },
+  };
+
+  const source: Record<string, unknown> = {
+    project: instance.project,
+    type: "copy",
+    source: `${instance.name}/${snapshot.name}`,
+    base_image: snapshot.config["volatile.base_image"] ?? "",
+  };
+
+  if (snapshot.stateful && values.stateful) {
+    basePayload["stateful"] = true;
+    source["live"] = false;
+  }
+
+  return {
+    ...basePayload,
+    source,
+  };
+};
+
+const CreateInstanceFromSnapshotForm: FC<Props> = ({
+  instance,
+  snapshot,
+  close,
+}) => {
+  const toastNotify = useToastNotification();
+  const { data: settings } = useSettings();
+  const isClustered = isClusteredServer(settings);
+  const controllerState = useState<AbortController | null>(null);
+  const eventQueue = useEventQueue();
+
+  const { data: projects = [], isLoading: projectsLoading } = useQuery({
+    queryKey: [queryKeys.projects],
+    queryFn: fetchProjects,
+  });
+
+  const { data: storagePools = [], isLoading: storagePoolsLoading } = useQuery({
+    queryKey: [queryKeys.storage],
+    queryFn: () => fetchStoragePools(instance.project),
+  });
+
+  const { data: clusterMembers = [], isLoading: clusterMembersLoading } =
+    useQuery({
+      queryKey: [queryKeys.cluster],
+      queryFn: fetchClusterMembers,
+      enabled: isClustered,
+    });
+
+  const { data: instances = [] } = useQuery({
+    queryKey: [queryKeys.instances],
+    queryFn: () => fetchInstances(instance.project),
+  });
+
+  const notifySuccess = (instanceName: string, instanceProject: string) => {
+    const instanceLink = (
+      <Link to={`/ui/project/${instanceProject}/instance/${instanceName}`}>
+        {instanceName}
+      </Link>
+    );
+
+    const message = <>Created Instance {instanceLink}.</>;
```

Comment:
```suggestion
    const message = <>Created instance {instanceLink}.</>;
```
I missed this in the other PRs. We should only use "Instance" if it is at the beginning of a notification. Could you please fix the other occasions in `UploadInstanceBtn` and `DuplicateInstanceForm` as well?

Reply:
fixed them all